### ISSUE TITLE
Uris with custom schemes are not working in 2.5.2.

### ISSUE THREAD

**Issue by pavinan** [IdentityServer/IdentityServer4#3535]
This update broke custom scheme urls. (Ex: "capacitor://localhost", "ionic://localhost"). Ionic app in ios or android renders pages with those urls inside webview.
According to [oauth rfc 6749 section 3.1.2](https://tools.ietf.org/html/rfc6749#section-3.1.2) custom schemes are allowed.

### Issue / Steps to reproduce the problem

Add some CORS origins with custom schemes. (Ex: "capacitor://localhost", "ionic://localhost")

### Relevant parts of the log file

```
Invalid client configuration for client sampleclient: AllowedCorsOrigins contains invalid origin: ionic://localhost
```

**Comment by brockallen** [comment]
>  This update broke custom scheme urls. (Ex: "capacitor://localhost", "ionic://localhost"). Ionic app in ios or android shows page in those urls.

Why are you putting those in `AllowsCorsOrigins`? They don't need to be in there.

**Comment by pavinan** [comment]
No, ionic creates some kind of server inside mobile and opens it in webview. That URL happens to be "capacitor://localhost".

Docs: https://ionicframework.com/docs/faq/cors

**Comment by adozenlines** [comment]
Still an issue in version 3.0.0 with custom schemes.

**Comment by adozenlines** [comment]
Mobile Applications that perform OAuth have the ability to register custom URL schemes. When such a scheme is specified in the call back URL of the request, the identity server doesn't redirect to the custom scheme and as a result, the mobile application never receives the access token.

**Comment by pavinan** [comment]
@adozenlines what framework are you using? Is it Ionic?

**Comment by pavinan** [comment]
Hi @brockallen this issue is blocking us from updating to .Net Core 3.0. Is there any alternative and can you check PR #3559. Thank you.

**Comment by pavinan** [comment]
@brockallen I have overridden ValidateAllowedCorsOriginsAsync in DefaultClientConfigurationValidator. Everything is working good.

**Comment by brockallen** [comment]
Tracked and fixed here:  #3880
